main.c

```diff
@@ -246,12 +253,37 @@ int main(int argc, const char *argv[])
 	progname_conv = strdup(path);
 	progname      = basename_internal(progname_conv);
 #endif /* CONFIG_APPELFLOADER_VFSEXEC */
+
 	/* Cut off kernel name (argv[0]) from argument vector */
 	argv = &argv[1];
 	argc -= 1;
 
 #endif /* !CONFIG_APPELFLOADER_CUSTOMAPPNAME */
 
+#if CONFIG_APPELFLOADER_VFSEXEC_ENVPATH
+	env_path = getenv("PATH");
+	if (env_path) {
+		realpath = locate_exec(path, env_path);
+		if (PTR2ERR(realpath) == -EINVAL) {
+			realpath = NULL;
+		} else if (PTRISERR(realpath) && PTR2ERR(realpath) != -EINVAL) {
+			uk_pr_err("%s: Failed to find executable in envirenment ($PATH): %s (%d)",
+				  progname, strerror(-PTR2ERR(realpath)),
+				  PTR2ERR(realpath));
+			goto out;
+		}
+	}
+#endif /* CONFIG_APPELFLOADER_VFSEXEC_ENVPATH */
+
+#if CONFIG_LIBPOSIX_PROCESS_EXECVE
+	ret = execve(realpath ? realpath : path,
+		     (char * const *)argv, (char * const *)environ);
+	if (unlikely(ret)) {
+		uk_pr_err("Could not execve (%d)\n", ret);
+		goto out;
```

Comment:
Now that I think you need to free `realpath` as well. THis is what `locate_exec` says.
```
/*
 * Routine that locates an executable in a colon-separated list of directories.
 * On success, it returns a malloc'ed C-string containing the full path. It is
 * in the responsibility of the caller to free the string after use with
 * `free()`.
 */
```

It doesn't look like it would be freed on the error path.

Reply:
Yes, with the reordering these got messed up, I'll have it fixed in the next revision.